libraries/chain/block_log.cpp

```diff
@@ -1133,50 +1136,44 @@ namespace eosio { namespace chain {
          fclose(ind_in);
    }
 
+   uint64_t trim_data::block_index(uint32_t n) const {
+      using namespace std;
+      EOS_ASSERT( n <= last_block, block_log_exception,
+                  "cannot seek in ${file} to block number ${b}, block number ${last} is the last block",
+                  ("file", index_file_name.string())("b",n)("last",last_block) );
+      return sizeof(uint64_t) * (n - first_block);
+   }
+
    uint64_t trim_data::block_pos(uint32_t n) {
       using namespace std;
-      index_pos = sizeof(uint64_t) * (n - first_block);
+      // can indicate the location of the block after the last block
+      if (n == last_block + 1) {
+         return ftell(blk_in);
+      }
+      const uint64_t index_pos = sizeof(uint64_t) * (n - first_block);
```

Comment:
Why not use `block_index(n)` instead?

Reply:
because I created the method and then forgot to swap out the code here :(